comments/services/feed.py

```diff
@@ -27,6 +28,11 @@ def get_comments_feed(
     else:
         qs = qs.filter(is_private=False)
 
+    if not include_deleted:
+        qs = qs.filter(
+            Q(is_soft_deleted=False) | Q(child_comments__is_soft_deleted=False)
```

Comment:
I assume this construction should be faster
```
from django.db.models import Exists, OuterRef
from comments.models import Comment

Exists(Comment.objects.filter(root_id=OuterRef("pk"), is_soft_deleted=False))
```

This also allows you not to use `distinct` here.

Could you please compare these queries performance against post with the biggest amount of comments?

Reply:
I evaluated the different in the query times between these two methods and found that for the question with the largest number of comments (post 11245 about who will become president has >4000 comments on it), running the queries 1000 times resulted in the average time of the original being `0:00:00.117956` while the average time of your revamp being `0:00:00.164822`

Both are trivially small in a way that I don't really care which one we choose, but technically the way I have it including the `distinct` call is faster.

I'm thus planning to leave it as it is.